carnival/carnival.ts

```diff
@@ -0,0 +1,1017 @@
+        let left = (screen.width >> 1) - (width >> 1) + 1;
+        let color1 = timerState.fontColor;
+        let color2 = timerState.bgColor;
+
+        screen.fillRect(left - 3, 0, width + 6, font.charHeight + 3, timerState.borderColor)
+        screen.fillRect(left - 2, 0, width + 4, font.charHeight + 2, color2)
+
+        if (seconds < 60) {
+            const top = 1;
+            const remainder = Math.idiv(millis % 1000, 10);
+
+            screen.print(formatDecimal(seconds) + ".", left, top, color1, font)
+            const decimalLeft = left + 3 * font.charWidth;
+            screen.print(formatDecimal(remainder), decimalLeft, top + 2, color1, smallFont)
+        }
+        else {
+            const minutes = Math.idiv(seconds, 60);
+            const remainder = seconds % 60;
+            screen.print(formatDecimal(minutes) + ":" + formatDecimal(remainder), left, 1, color1, font);
+        }
+    }
+
+    function formatDecimal(val: number) {
+        val |= 0;
+        if (val < 10) {
+            return "0" + val;
+        }
+        return val.toString();
+    }
+
+    function timerHUD() {
+        if (timerState) return;
+
+        timerState = new TimerState();
+
+        scene.createRenderable(
+            scene.HUD_Z,
+            () => {
+
+                // show timer
+                if (timerState.visibilityFlag & info.Visibility.Countdown) {
+                    const scene = game.currentScene();
+                    //const elapsed = scene.millis();
+                    const elapsed = control.timer1.millis();
+                    drawTimer(elapsed);
+                    let t = elapsed / 1000;
+                    if (t <= 0) {
+                        t = 0;
+
+                    }
+                }
+            }
+        );
+    }
+
+
+
+    /**
+     * Adds game end style to countdown
+     */
+    //% color="#b70082"
+    //% group="Countdown"
+    //% blockId=start_countdown_game
+    //% block="start countdown $myTime (s) and game over $winType || effect $winEffect"
+    //% myTime.defl=15
+    //% winType.defl=winTypes.Score
+    //% winEffect.defl=effects.confetti
+    //% inlineInputMode=inline
+    //% help=github:carnival/docs/start_countdown_game
+    export function startCountdownGame(myTime: number, winType: winTypes, winEffect?: effects.BackgroundEffect) {
+        if (!winType)
+            winType = winTypes.Win;
+        if (!winEffect && winType != winTypes.Lose) {
+            winEffect = effects.confetti;
+        }
+        else { winEffect = effects.melt; }
+
+        init(winType, winEffect);
+        info.startCountdown(myTime);
+
+    }
+
+    function saveLowScore(newLow: number) {
+        const curr = settings.readNumber("low-score")
+        if (curr == undefined || newLow < curr) {
+            settings.writeNumber("low-score", newLow);
+        }
+    }
+
+    // Get the last recorded low score
+    function lowScore(): number {
+        return settings.readNumber("low-score");
```

Comment:
is this being used? it looks like you just read the number in place below instead of calling this function

Reply:
Deleted…though I could see an argument for having it for other functions to use.  They can add it when they need it :)